Cython/Compiler/Tests/TestBuffer.py

```diff
@@ -49,13 +49,13 @@ def nonfatal_error(self, error):
         self.assertTrue(self.expect_error)
 
     def parse_opts(self, opts, expect_error=False):
-        assert opts != ""
+        self.assertNotEqual(opts, "")
         s = u"def f():\n  cdef object[%s] x" % opts
         self.expect_error = expect_error
         root = self.fragment(s, pipeline=[NormalizeTree(self), PostParse(self)]).root
         if not expect_error:
             vardef = root.stats[0].body.stats[0]
-            assert isinstance(vardef, CVarDefNode)  # use normal assert as this is to validate the test code
+            self.assertIsInstance(vardef, CVarDefNode)
```

Comment:
I can't figure out why we have to use normal assert here.

Reply:
> Because it is not a test but an assertion about the test code itself. Like an assert in normal production code.

@scoder That doesn't mean we can't use `unittest`'s `assert` for this part, does it?
In case of problems, the error will still occur, but we will have a more beautiful and understandable message about it in our hands.